releasenotes/notes/force-http1-for-log-agent-2cbfba763697ab42.yaml

```diff
@@ -0,0 +1,15 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+     Introduced a new configuration variable logs_config.transport_type, allowing users to enforce HTTP/1.1 for outgoing HTTP connections in the Datadog Agent. This provides better control over transport protocols, improving compatibility with systems that do not support HTTP/2.
```

Comment:
Maybe we should reword this to highlight that it changes the default behavior to use HTTP2 in most cases?

Reply:
But is it? Isn't the http.Client configuring for HTTP2 by default nowadays?

Reply:
I dove into this a bit and from what i found:

If we provide custom Dial, DialTLS, DialContext functions, or a custom TLSClientConfig (in which case we do [here](https://github.com/DataDog/datadog-agent/blob/log-agent-http/2/pkg/util/http/transport.go#L108-L122)), HTTP/2 is disabled by default **unless** configured for via:
1. External library: `http2.ConfigureTransport` [ This is better than 2nd option since it accounts for fallback to closest available protocol ]
2. Force attempt http2 via transport: `transport.ForceAttemptHTTP2 = true`

Reply:
Oh that's interesting! Thank you